### PR TITLE
Add api for song

### LINKED ISSUE #3: Create API and test for songs
- Create the methods for the controller
- Create the rspec for the controller
- Add validations if it's necessary
- Configure the routes

### PR DESCRIPTION
**Pull request by Linzeur**
Add validation in model
Add all methods for api
Add all test cases for api
Add routes
Resolve #3 issue

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -2,7 +2,25 @@
...
+
+    resources :songs, only: [ :index, :show] do
+      member do
+        get :artists
+        get :albums
+
+        put "/progress" => :song_progress
+        put "/rating" => :song_rating 
+      end
+      get "/search" => :search, on: :collection
```
Comment (CesarCachay) [inline config/routes.rb]: This will show us the complete list of songs right?

Comment (Linzeur) [reply config/routes.rb]: I asked Kattya yesterday and the purpose is when you call to the search route, you pass a parameter which it search all songs by title

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,2 +1,2 @@
-class ApplicationController < ActionController::API
+class ApplicationController < ActionController::Base
```
Comment (Gatts1) [inline app/controllers/application_controller.rb]: What does this line do exactly?

Comment (CesarCachay) [reply app/controllers/application_controller.rb]: I think it allows to render the other namespaces not only "api" because we also have "admin"

**Pull request merged**